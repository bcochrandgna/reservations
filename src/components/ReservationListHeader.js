// @flow
import React from 'react';
import { View, Text } from 'react-native';
import { Query } from 'react-apollo';
import GET_RESERVATIONS_COUNT from '../data/queries/getReservationsCount';
import i18n from '../helpers/i18n';
import styles from '../styles/components/ReservationListHeader';
import ReservationSortButton from './ReservationSortButton';

const ReservationListHeader = () => (
  <View style={styles.container}>
    <Query query={GET_RESERVATIONS_COUNT}>
      {({ data, error, loading }) => {
        const count = loading || error ? 0 : data.reservationsConnection.aggregate.count;
        return <Text style={styles.titleText}>{`${i18n.t('reservations')} (${count})`}</Text>;
      }}
    </Query>
    <ReservationSortButton />
  </View>
);

export default ReservationListHeader;
