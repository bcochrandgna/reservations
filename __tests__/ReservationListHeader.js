import React from 'react';
import { shallow } from 'enzyme';
import ReservationListHeader from '../src/components/ReservationListHeader';

describe('ReservationListHeader', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<ReservationListHeader />);
    expect(wrapper).toMatchSnapshot();
  });
});
