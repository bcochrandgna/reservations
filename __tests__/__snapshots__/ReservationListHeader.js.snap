// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationListHeader renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 5,
    }
  }
>
  <Query
    query={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": undefined,
            "operation": "query",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "reservationsConnection",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "aggregate",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "count",
                              },
                              "selectionSet": undefined,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 90,
          "start": 0,
        },
      }
    }
  >
    <Component />
  </Query>
  <ReservationSortButton />
</View>
`;
